internal/rpc/user_devices.go

```diff
@@ -762,86 +760,63 @@ func (s *userDeviceRPCServer) ClearMetaTransactionRequests(ctx context.Context,
-		).One(ctx, s.dbs().Reader)
-		if err != nil {
-			return nil, err
-		}
-
-		if apiInt.R.UserDevice == nil {
-			return nil, fmt.Errorf("failed to find user device %s for integration %s", deleteRequest.UserDeviceId, deleteRequest.IntegrationId)
-		}
+	apiInt, err := models.UserDeviceAPIIntegrations(
+		models.UserDeviceAPIIntegrationWhere.UserDeviceID.EQ(req.UserDeviceId),
+		models.UserDeviceAPIIntegrationWhere.IntegrationID.EQ(req.IntegrationId),
+		qm.Load(models.UserDeviceAPIIntegrationRels.UserDevice),
+	).One(ctx, s.dbs().Reader)
+	if err != nil {
+		log.Err(err).Msg("failed to retrieve integration")
+		return nil, err
+	}
 
-		dd, err := s.deviceDefSvc.GetDeviceDefinitionByID(ctx, apiInt.R.UserDevice.DeviceDefinitionID)
-		if err != nil {
-			return nil, fmt.Errorf("deviceDefSvc error getting device definition by id %s: %w", apiInt.R.UserDevice.DeviceDefinitionID, err)
-		}
+	if apiInt.R.UserDevice == nil {
+		log.Info().Msg("failed to find user device")
+		return nil, fmt.Errorf("failed to find user device %s for integration %s", req.UserDeviceId, req.IntegrationId)
+	}
 
-		integ, err := s.deviceDefSvc.GetIntegrationByID(ctx, deleteRequest.IntegrationId)
-		if err != nil {
-			return nil, fmt.Errorf("deviceDefSvc error getting integration by id %s: %w", deleteRequest.IntegrationId, err)
-		}
+	integ, err := s.deviceDefSvc.GetIntegrationByID(ctx, req.IntegrationId)
+	if err != nil {
+		return nil, fmt.Errorf("deviceDefSvc error getting integration by id %s: %w", req.IntegrationId, err)
+	}
 
-		switch integ.Vendor {
-		case constants.SmartCarVendor:
-			if apiInt.TaskID.Valid {
-				err = s.smartcarTaskSvc.StopPoll(apiInt)
-				if err != nil {
-					return nil, fmt.Errorf("failed to stop smartcar poll: %w", err)
-				}
+	switch integ.Vendor {
+	case constants.SmartCarVendor:
+		if apiInt.TaskID.Valid {
+			err = s.smartcarTaskSvc.StopPoll(apiInt)
+			if err != nil {
+				log.Err(err).Msg("failed to stop smartcar poll")
+				return nil, fmt.Errorf("failed to stop smartcar poll: %w", err)
 			}
-		case constants.TeslaVendor:
-			if apiInt.TaskID.Valid {
-				err = s.teslaTaskService.StopPoll(apiInt)
-				if err != nil {
-					return nil, fmt.Errorf("failed to stop tesla poll: %w", err)
-				}
+		}
+	case constants.TeslaVendor:
+		if apiInt.TaskID.Valid {
+			err = s.teslaTaskService.StopPoll(apiInt)
+			if err != nil {
+				log.Err(err).Msg("failed to stop tesla poll")
+				return nil, fmt.Errorf("failed to stop tesla poll: %w", err)
 			}
 		}
+	default:
+		log.Info().Str("vendor", integ.Vendor).Msg("stop user integration poll not implemented for vendor")
+		return nil, fmt.Errorf("stop user integration poll not implemented for vendor %s", integ.Vendor)
+	}
 
-		_, err = apiInt.Delete(ctx, s.dbs().Reader)
-		if err != nil {
-			return nil, fmt.Errorf("failed to delete integration: %w", err)
-		}
+	apiInt.Status = models.UserDeviceAPIIntegrationStatusAuthenticationFailure
+	apiInt.TaskID = null.String{}
```

Comment:
I don't think we should remove the TaskID, it can still be used for reference. I think changing the status is good enough.